src/mlpack/bindings/cli/add_to_po.hpp

```diff
@@ -14,95 +13,171 @@
 #define MLPACK_BINDINGS_CLI_ADD_TO_PO_HPP
 
 #include <mlpack/core/util/param_data.hpp>
-#include <boost/program_options.hpp>
 #include <mlpack/core/util/is_std_vector.hpp>
 #include "map_parameter_name.hpp"
 
+#include "third_party/CLI/CLI11.hpp"
+
 namespace mlpack {
 namespace bindings {
 namespace cli {
 
 /**
- * Add a non-vector option to boost::program_options.
- *
- * @param boostName The name of the option to add to boost::program_options.
+ * Add a tuple option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
-             const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
-             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0)
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
+             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0,
+             const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+             const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+             const typename boost::enable_if<std::is_same<T,
+                std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(), boost::program_options::value<T>(),
-      descr.c_str());
+  app.add_option_function<T>(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
 }
 
 /**
- * Add a vector option to boost::program_options.  This overload will use the
- * multitoken() option.
- *
- * @param boostName The name of the option to add to boost::program_options.
+ * Add a serializable option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
+ * @param descr Description string for parameter.
+ * @param desc Options description to add parameter to.
+ */
+template<typename T>
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
+             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0,
+             const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+             const typename boost::enable_if<data::HasSerialize<T>>::type* = 0,
+             const typename boost::disable_if<std::is_same<T,
+                std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
+{
+  app.add_option_function<T>(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
+}
```

Comment:
This one applies for a serializable model, so if I look at the `SetParam()` implementation there, we can adapt this in a similar way:

```
  app.add_option_function<T>(cliName.c_str(),
      [&param](const T& value)
      {
        typedef std::tuple<T*, typename ParameterType<T>::type> TupleType;
        TupleType& tuple = *boost::any_cast<TupleType>(&param.value);
        std::get<1>(tuple) = boost::any_cast<std::string>(value);
      },
      param.desc.c_str());
```

Note that we hold a `T*` here, because we stripped the pointer off in the top-level call with the `std::remove_pointer`.